model/src/gateway/event/close_code.rs

```diff
@@ -0,0 +1,47 @@
+use serde_repr::{Deserialize_repr, Serialize_repr};
+
+/// Gateway close event codes.
+#[derive(
+    Clone, Copy, Debug, Deserialize_repr, Eq, Hash, PartialEq, PartialOrd, Ord, Serialize_repr,
+)]
+#[repr(u16)]
+pub enum CloseCode {
```

Comment:
This probably deserves to go in just `gateway`, since it's not an event and is related to the state of the Gateway WebSocket connection itself.